### PR TITLE
Add `GF2AddK` for in place addition of a constant over GF($2^m$)

### PR DESCRIPTION
**Pull request by tanujkhattar**
Part of a larger effort to support galois field arithmetic in qualtran. xref https://github.com/quantumlib/Qualtran/issues/1419

Currently on top of https://github.com/quantumlib/Qualtran/pull/1442

### CODE REVIEW

File: qualtran/bloqs/gf_arithmetic/gf2_add_k.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+@attrs.frozen
+class GF2AddK(Bloq):
+    r"""In place addition of a constant $k$ for elements in GF($2^m$).
+
+    The bloq implements in place addition of a classical constant $k$ and a quantum register
+    $|x\rangle$ storing elements from GF($2^m$). Addition in GF($2^m$) simply reduces to a component
+    wise XOR, which can be implemented via X gates.
+
+        $$
+        |x\rangle  \rightarrow |x + k\rangle
```
Comment (mpharrigan) [inline qualtran/bloqs/gf_arithmetic/gf2_add_k.py:38]: don't indent

File: qualtran/bloqs/gf_arithmetic/gf2_add_k.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        return is_symbolic(self.k, self.bitsize)
+
+    def build_composite_bloq(self, bb: 'BloqBuilder', *, x: 'Soquet') -> Dict[str, 'Soquet']:
+        if self.is_symbolic():
+            raise DecomposeTypeError(f"Cannot decompose symbolic {self}")
+        xs = bb.split(x)
+
+        for i, bit in enumerate(self._bits_k):
+            if bit == 1:
+                xs[i] = bb.add(XGate(), q=xs[i])
```
Comment (mpharrigan) [inline qualtran/bloqs/gf_arithmetic/gf2_add_k.py:76]: `XorK`?

**Review by mpharrigan — APPROVED** [review]

File: qualtran/bloqs/gf_arithmetic/gf2_add_k.py (hunk 2 above)
Comment (tanujkhattar) [reply qualtran/bloqs/gf_arithmetic/gf2_add_k.py:76]: `XorK` assumes the `K` is an `int`. This bloq is the GF2 equivalent of a `XorK`, similar to how `GF2Add` is equivalent to `Xor`.

File: qualtran/bloqs/gf_arithmetic/gf2_add_k.py (hunk 1 above)
Comment (tanujkhattar) [reply qualtran/bloqs/gf_arithmetic/gf2_add_k.py]: done

**Pull request merged**